tsdoc/src/parser/DocCommentAssembler.ts

```diff
@@ -28,37 +31,89 @@ export class DocCommentAssembler {
 
-      let pruneNode: boolean = false;
+      let skipNode: boolean = false;
 
       switch (docNode.kind) {
         case DocNodeKind.BlockTag:
           const docBlockTag: DocBlockTag = docNode as DocBlockTag;
           // Do we have a definition for this tag?
           const tagDefinition: TSDocTagDefinition | undefined
-            = this._configuration.tryGetTagDefinitionUpperCase(docBlockTag.tagNameForComparisons);
+            = this._configuration.tryGetTagDefinitionWithUpperCase(docBlockTag.tagNameWithUpperCase);
           if (tagDefinition) {
-            if (tagDefinition.syntaxKind === TSDocTagSyntaxKind.ModifierTag) {
-              modifierTagSet.addModifierTag(docBlockTag);
-              pruneNode = true;
+            switch (tagDefinition.syntaxKind) {
+              case TSDocTagSyntaxKind.BlockTag:
+                // This is a block tag, so start a new block
+                currentBlock = new DocBlock({
+                  blockTag: docBlockTag
+                });
+                skipNode = true;
+                blocks.push(currentBlock);
+                break;
+              case TSDocTagSyntaxKind.ModifierTag:
+                // The block tag was recognized as a modifier, so add it to the modifier tag set
+                modifierTagSet.addModifierTag(docBlockTag);
+                skipNode = true;
+                break;
             }
           }
           break;
-        default:
-          break;
       }
-      if (!pruneNode) {
-        prunedDocNodes.push(docNode);
+
+      if (!skipNode) {
+        if (currentBlock) {
+          currentBlock.appendNode(docNode);
+        } else {
+          summaryNodes.push(docNode);
+        }
+      }
+    }
+
+    // TODO: If there is no "@remarks" block, then we could treat the first non-trivial DocParagraph
+    // as the summary, and the rest as the remarks.
+    this._docComment.summarySection.appendNodes(summaryNodes);
+    return blocks;
+  }
+
+  private _arrangeBlocks(blocks: DocBlock[]): void {
+    // Now sift the blocks into normal and "custom" blocks
+    for (const block of blocks) {
+      switch (block.blockTag.tagNameWithUpperCase) {
+        case CoreTags.remarks.tagNameWithUpperCase:
+          this._docComment.remarksBlock = block;
+          break;
+          case CoreTags.param.tagNameWithUpperCase:
+          this._docComment.paramBlocks.push(this._constructParamBlock(block));
+          break;
+        case CoreTags.returns.tagNameWithUpperCase:
+          this._docComment.returnsBlock = block;
+          break;
+        default:
+          this._docComment.appendCustomBlock(block);
       }
     }
+  }
 
-    this._docComment.remarks.appendNodes(prunedDocNodes);
+  private _constructParamBlock(block: DocBlock): DocParamBlock {
+    const paramBlock: DocParamBlock = new DocParamBlock({
+      excerpt: block.excerpt,
+      parameterName: block.blockTag.tagName
+    });
+    paramBlock.appendNodes(block.nodes);
```

Comment:
>appendNodes(block.nodes); [](start = 15, length = 25)

here is where you would parse variable name from text

Reply:
I decided to address this in a separate PR, because it needs some additional parser machinery unpack and repack the node.

---
In reply to: [210144981](https://github.com/Microsoft/tsdoc/pull/42#discussion_r210144981) [](ancestors = 210144981)